src/destinations/dune.py

```diff
@@ -26,9 +34,19 @@ class DuneDestination(Destination[TypedDataFrame]):
 
     """
 
-    def __init__(self, api_key: str, table_name: str, request_timeout: int):
+    def __init__(
+        self,
+        api_key: str,
+        table_name: str,
+        request_timeout: int,
+        insertion_type: InsertionPolicy = "append",
+    ):
         self.client = DuneClient(api_key, request_timeout=request_timeout)
+        if "." not in table_name:
+            raise ValueError("Table name must be in the format namespace.table_name")
```

Comment:
That the end to end test is expected to fail because it is reading a remote file from main. Once this PR lands. The file will be updated and the e2e test will pass again.